src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1131,16 +837,93 @@ PrimaryNodeForGroup(int32 groupId, bool *groupContainsNodes)
 
 
 /*
- * ActivateNodeList iterates over the nodeList and activates the nodes.
- * Some part of the node activation is done parallel across the nodes,
- * such as syncing the metadata. However, reference table replication is
- * done one by one across nodes.
+ * DeactivateMetadataNodesInSeparateTransaction unsets isactive and metadatasynced flags
+ * in separate connection to localhost.
+ */
+static void
+DeactivateMetadataNodesInSeparateTransaction(MetadataSyncContext *context)
+{
+	Assert(context->transactionMode == METADATA_SYNC_NON_TRANSACTIONAL);
+	Assert(!MetadataSyncCollectsCommands(context));
+
+	/*
+	 * set flags to false for all activated nodes to prevent an inconsistent
```

Comment:
lets not set `isactive`, because that could break the traffic on that node.

I know logically we should be marking the node as not active, but the "active" node terminology is complicated.